[Mod] Features requested for kick/ban.

Ianardo-DiCaprio opened the pull request:
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes
Added features to kick and ban requested in #2649 . This will now send an embed to the user when they are kicked/banned from a server detailing if it was a kick or ban, from what server and the reason for the kick/ban.

Drapersniper committed:
Sigh i removed this before ...

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Reorder dispatch signatures so all 3 new dispatch have matching signature

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Formatting

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Edited Error Event to support localtracks

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fix a Crash on track crash :awesome:

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Yikes soo much spam

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Remove spam and improve existance check

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Repeat and Auto-play are mutually exclusive now

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
DEBUGS for Preda

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Vimeo tracks can be from both these domains "vimeo.com", "beam.pro"

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
I mean Mixer can be from those 2 domains ....

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fixed `search sc` command

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Run everything though lints.
rename localtracks module to dataclasses
Clear lock on errors

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Try to speed up long playlist loading

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Im an idiot

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Im an idiot

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Added logging for writes

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Fix crash on cog reload

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Fix for runtimewarning ?

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Fix for Local Track cache

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Remove broken tracks from queue on exception
Theoretically do not auto play if track stop reason is Stopped or cleanup

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Previous commit was a fluke ... ignore it

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Change from cleanup to Replaced

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Fixed AttributeError: 'Track' object has no attribute 'info'.
[p]skip will only work for autoplay is there a track being played.
Fixed Console spam if query saving failed in the background while reloading bot.
Autoplay now respect [p]stop command

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Black formatting
Fix Issue with auto play working when there is songs in the queue
Stop notifying queue ended if autoplay is on

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fixed a crash on track load timeout

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
[p]playlist start will now show the playlist name in embed body
Improved Logic for handling broken tracks when repeat is on.

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Enqueue tracks as soon as we have the youtube URL ....

This basically changes how spotify urls are handled

Need to test saving spotify playlist
Need to test loading a spotify playlist from file
Need to test enqueuing a spotify playlist

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Updated a track whrn enqueuing spotify playlist

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Debug

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Debug

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Debug

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Debug

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Debug

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Debug

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Debug

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Debug

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Debug

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Debug

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Revert spotify_enqueue changes

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Merge remote-tracking branch 'origin/audiochanges' into audiochanges

Drapersniper committed:
Revert spotify_enqueue changes

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Allow to set Lavalink jar version from Environment vars

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Allow to set Lavalink jar version from Environment vars

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Fix for a crash on Equalizer, Merge Spotify_enqueue changes and revert manager changes

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Break playlist enqueue after 10 consecutive failures

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Auto DC, is not compatible with Auto Play

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Make notifier aware of guild its being called for

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Type checking

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Remove lock from 2 exits that i didn't before

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Fixed TypeError: spotify_enqueue() got an unexpected keyword argument 'notify'

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Reorder toggles to alphabetical order

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Update Query to handle spotify URIs

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
update database

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Merge branch 'V3/develop' into audiochanges

Drapersniper committed:
Dont say tracks enqued on invalid link
Make autop lay a mod only setting

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Merge remote-tracking branch 'origin/audiochanges' into audiochanges

Drapersniper committed:
Dont say tracks enqued on invalid spotify link

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Set default age to 365 days

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Allow Audio mods to set auto play playlists.
Save playlists songs to cache when migrating

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Black formatting

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
[p]eq cooldown is not triggered is player check fails (i.e if nothing is currently playing)
Adding and removing reaction is no longer a blocking action

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
changelog for non blocking reaction handles

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Show auto dc  and auto play settings by default

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
lint is being a bitch

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
lint changes

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
stop caching local tracks

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
List of Lavalink.Tracks natively added to Playlist Objects

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Fix UX changes and should fix autoplay

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Fixed Skip x number of tracks

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Lint changes

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Remvoe dead code

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Update playlist embed formatting to reflect Preda's suggestions

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Update change logs

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Add `async with ctx.typing():` to queue and to local folder

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Stop queuing now when queue is empty with [p]queue

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
fix ctx.typing()

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
fix ctx.typing()

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Part 1

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Dont check local track author and name if title is Unknown

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Makes auto play more random

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fixes local play
Fixed missing format

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Query.process_input accept lavalink.Track objects

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
docstrings

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Add TODO for timestamp support

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Improve autoplay from cache logic (possibly slightly slower but more efficient overall)

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Add My Lavalink PR as a dependency
Remember to remove this .... The PR will bump it to 0.3.2

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Add My Lavalink PR as a dependency
Remember to remove this .... The PR will bump it to 0.3.2

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Add My Lavalink PR as a dependency
Remember to remove this .... The PR will bump it to 0.3.2

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Compile all regex at runtime

Signed-off-by: Draper <guyreis96@gmail.com>

Drapersniper committed:
Merge branch 'V3/develop' into audiochanges

Drapersniper committed:
Fixes local play
Fixed missing format

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Merge remote-tracking branch 'origin/audiochanges' into audiochanges

Drapersniper committed:
Revert Dep error

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
black

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fixed attribute error

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
add `self.bot.dispatch("audio_disconnect", ctx.guild)` dispatch when the player is disconnected

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Removed shuffle lock on skip

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Better logic for auto seek (timestamps)

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Better logic for auto seek (timestamps)

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fixes timestamps on spotify tracks

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Add ctx typing to playlist enqueue

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fix Deps

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Black formatting + Using new lavalink methods for shuffling

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
remove ctx.typing from playlist start

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fixes typerror when enqueuing spotify playlists

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fix keyerror

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
black formatting, + embed for [p]audioset cache as I forgot it before

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fix Error on playlist upload

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fix Text help for bump

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Allow track bumping while shuffle is on

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Edit bump embed to be consistent with other embed
Hyperlink tracks and removed dynamic title

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Black

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Errors not printing fix?

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Errors not printing fix?

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Track enqueued footer now shows correct track position when shuffle is on

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Update changelogs

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fix is_owner check in audioset settings

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Changelogs

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Dont store searches with no results in cache, fix malformated playlist to cache upon settings migration

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
_clear_lock_on_error > Needs to be reviewed to see if it has been done correctly

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
_clear_lock_on_error > Needs to be reviewed to see if it has been done correctly

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fix Query search so that it works with absolute paths for localtracks

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Extra error if lavalink is set to external and  the query is a localtrack and nothing is found

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Black

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
More detailed error message

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
[p]seek and [p]skip can be used by user if they are the song requester while DJ mode is enabled, if votes are disabled. , [p]queue shuffle can be used to shuffle the queue manually. and [p]queue clean self can be used to remove all songs you requested from the queue.

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
black

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
All the fixes + a `should_auto_play` dispatch for the tech savy peeps

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Spellchecker + Pythonic changes

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
NO spam for logs

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Pass Current voice channel to `red_audio_should_auto_play` dispatch

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Black

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
playlist upload also updates cache in the background

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
playlist upload also updates cache in the background

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Add scope to playlist picker

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Delete Playlist picker message once something is selected

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
OCD Fix

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Facepalm

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fix a Potential crash

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Update my stupidity

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Merge branch 'V3/develop' into audiochanges

Drapersniper committed:
Auto Pause +  Skip tracks already in playlist upon playlist append + a command to remove duplicated tracks from playlist

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fix DJ mode when Role is deleted - Credits go to Neuro Assassin#4779
Fix an issue where auto play MAY not trigger

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Change log to  Neuro Assassin#4779 fix

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Black

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Dont auto pause manual pauses

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Adds `[p]autoplay` that can be run by mods or higher

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
:facepalm:

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
2x :facepalm:

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fixed wrong import

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Added Autoplay notify

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Added Autoplay notify

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Black

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Store Track object as prev song instead of URI

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Black why do u hate me

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fix command name

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fix Autoplay notify

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fix missing await and TypeError, Thanks Flame

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Add a list of tracks to show as a menu

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
adds the `[p]genre` command which uses the Spotify and Youtube API

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Enqueue Playlists from genre command

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Pretify `[p]genre`

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fix a Typo and correct jukebox charge order

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Add genre command to error handling

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Type checking

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Update naming scheme for `[p]genre`

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Black why do you hate me

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fixed `[p]local start`
Playlist picker auto selects if theres just 1 playlist found
`[p]queue cleanself` added

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
*sigh* back compatibility with old localtrack paths

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
*sigh* back compatibility with old localtrack paths, even more

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
*sigh* back compatibility with old localtrack paths Even more

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fixes localtracks in playlist info command

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Debug Local Strings

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Debug Local Strings

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fixes `[p]playlist info` for local tracks + fixed error in `[p]remove`

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Black

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fixes formatting in `[p]playlist info`

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fix an issue with User Scope playlists were not being deleted

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Typechecking

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Black

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fix the logic of `delegate_autoplay`

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fix a Crash on Load due to type hinting

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fix a Crash on Load due to type hintingBlack + fix order of `red_audio_should_auto_play`

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Add `red_audio_initialized` dispatch so that ownership of auto play can be maintained after a reload

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Check if the current owner is loaded before raising an error

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fixes the Existence Check in `delegate_autoplay`

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Turns `own_autoplay` in a property of Audio and improves `delegate_autoplay` Thanks Sinbad!

Signed-off-by: Guy <guyreis96@gmail.com>

Drapersniper committed:
Fix for Localtracks playlists

Signed-off-by: Guy <guyreis96@gmail.com>

Ianardo-DiCaprio committed:
Initial Commit

Ianardo-DiCaprio committed:
Add changelog

Ianardo-DiCaprio committed:
Initial Commit

Ianardo-DiCaprio committed:
Added changelog

Ianardo-DiCaprio committed:
Initial Commit

Ianardo-DiCaprio committed:
Add files via upload

Drapersniper committed:
When disconnecting send `Disconnecting...`
Fix Stop after a skip
Fix UX discrepancy on Playlist IDs
Fixed Exception when theres a track error

Signed-off-by: Guy <guyreis96@gmail.com>

DevilXD commented on redbot/cogs/mod/kickban.py:
```diff
@@ -174,6 +174,30 @@ class KickBanMixin(MixinMeta):
+                reason = reason
+            self == self.bot.get_cog("Mod")
+            toggle = await self.settings.guild(guild).toggle_dm()
+            try:
```
This entire try-except is unnecessary if you're already using suppress inside.

DevilXD commented on redbot/cogs/mod/kickban.py:
```diff
@@ -174,6 +174,30 @@ class KickBanMixin(MixinMeta):
             return
         audit_reason = get_audit_reason(author, reason)
         try:
```
This try shouldn't really be here if you're sending the message before the kick, since there might be a situation where the message will send successfully, and then the kick itself will fail.
Since there already is a role hierarchy check on lines 163-171, and proper decorators for bot and user permissions are used, the `except` case with `discord.Forbidden` should be removed since it will never have a chance to run after the changes (if I'm wrong on this one, please correct me - hierarchy and permissions should be the only cause of Forbidden here, and these are covered).

In addition, the `try` statement should be moved to right before the kick happens, to make sure it's guarding the right / intended piece of code.

Jackenmen replied:
Unless I'm getting it wrong, kick can't fail because the command has a `@commands.bot_has_permissions(kick_members=True)` check.

DevilXD commented on redbot/cogs/mod/kickban.py:
```diff
@@ -213,11 +237,29 @@ class KickBanMixin(MixinMeta):
+            reason = reason
+        self == self.bot.get_cog("Mod")
+        toggle = await self.settings.guild(guild).toggle_dm()
+        try:
```
Same deal as before, unnecessary try-except with suppress already inside.

Ianardo-DiCaprio committed:
Update kickban.py

Ianardo-DiCaprio committed:
Update kickban.py

Ianardo-DiCaprio committed:
Add files via upload

Ianardo-DiCaprio committed:
Missed this when deleting moderator info.

Ianardo-DiCaprio committed:
Wrong indentation.

Drapersniper committed:
add `on_red_audio_unload` dispatch

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

DevilXD reviewed (dismissed):
Looks good to me now 👌

Ianardo-DiCaprio committed:
Remove unneeded else:

Jackenmen commented on redbot/cogs/mod/kickban.py:
```diff
@@ -173,6 +173,18 @@ class KickBanMixin(MixinMeta):
             await ctx.send(_("I cannot do that due to discord hierarchy rules"))
             return
         audit_reason = get_audit_reason(author, reason)
+        if reason == None:
```
`None` is singleton so you should use identity check
```suggestion
        if reason is None:
```

Jackenmen commented on redbot/cogs/mod/kickban.py:
```diff
@@ -173,6 +173,18 @@ class KickBanMixin(MixinMeta):
+            reason = "No reason was given."
+        self == self.bot.get_cog("Mod")
+        toggle = await self.settings.guild(guild).toggle_dm()
+        if toggle == True:
```
`True` is singleton so you should either use identity check or since it's a bool, you can just use `if toggle:`
```suggestion
        if toggle:
```

Jackenmen commented on redbot/cogs/mod/kickban.py:
```diff
@@ -213,11 +225,24 @@ class KickBanMixin(MixinMeta):
+            reason = "No reason was given."
+        self == self.bot.get_cog("Mod")
+        toggle = await self.settings.guild(guild).toggle_dm()
+        if toggle == True:
```
`True` is singleton so you should either use identity check or since it's a bool, you can just use `if toggle:`
```suggestion
        if toggle:
```

Jackenmen commented on redbot/cogs/mod/kickban.py:
```diff
@@ -213,11 +225,24 @@ class KickBanMixin(MixinMeta):
         Minimum 0 days, maximum 7. Defaults to 0."""
+        author = ctx.author
+        guild = ctx.guild
+        if reason == None:
```
`None` is singleton so you should use identity check
```suggestion
        if reason is None:
```

Jackenmen reviewed (commented):
Few small irks

Ianardo-DiCaprio committed:
Update redbot/cogs/mod/kickban.py

Co-Authored-By: jack1142 <6032823+jack1142@users.noreply.github.com>

Ianardo-DiCaprio committed:
Update redbot/cogs/mod/kickban.py

Co-Authored-By: jack1142 <6032823+jack1142@users.noreply.github.com>

Ianardo-DiCaprio committed:
Update redbot/cogs/mod/kickban.py

Co-Authored-By: jack1142 <6032823+jack1142@users.noreply.github.com>

Ianardo-DiCaprio committed:
Update kickban.py

Drapersniper committed:
Fix a crash on track start where `player.current` can be none?

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Missing new line

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Ianardo-DiCaprio committed:
Merge branch 'V3/develop' into patch-3-1

Ianardo-DiCaprio committed:
Merge branch 'V3/develop' into patch-2-1

Ianardo-DiCaprio committed:
Merge branch 'V3/develop' into patch-1

Drapersniper committed:
Allow `--author` for playlist to be used to filter playlist for an specific author.
Plus a few bugfixes for UX

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Rename `remdupe` to `dedupe`
Make global scope always be referenced as Global
add missing backwards quotes around the Playlist ID for 1 string

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Ianardo-DiCaprio committed:
Review changes

Drapersniper committed:
Towncrier entries for dep changes

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Remove track index when shuffle is on
Fix Progress bar for livestreams

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Trigger autoplay on `QUEUE_END` event instead of `TRACK_END`

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Ianardo-DiCaprio committed:
Changed to use contextlib.suppress

Ianardo-DiCaprio committed:
Update warnings.py

Ianardo-DiCaprio committed:
Merge pull request #13 from Cog-Creators/V3/develop

d

Drapersniper committed:
Can't reproduce Ians bug but here a safeguard agaisnt it just in case

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Fixes 2 Messages that had the wrong formatting

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
standerdize playlist naming scheme

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Fix `[p]autoplay` message when Notify is enabled

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
y u h8 me black

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Fix an issue with `[p]audioset localpath` where the localtracks folder was incorrect

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Pythonic formatting

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Ugh

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Fix a typo

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Ianardo-DiCaprio committed:
Merge pull request #21 from Cog-Creators/V3/develop

r

Drapersniper committed:
Silently try to delete messages + fixes error Ian found with `[p]genre`

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
sigh black

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Merge branch 'V3/develop' of https://github.com/Cog-Creators/Red-DiscordBot into audiochanges

Drapersniper committed:
Merge branch 'V3/develop' into audiochanges

Drapersniper committed:
Add humanize_number usage correctly

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Bump RLL to 0.4.0

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Ianardo-DiCaprio committed:
Merge pull request #22 from Cog-Creators/V3/develop

p

Drapersniper committed:
Update changelog entries

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Merge branch 'V3/develop' into audiochanges

Drapersniper committed:
Change `bot.db` to new API's added by #2967

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Additional reformatting

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Merge branch 'V3/develop' into audiochanges

Drapersniper committed:
Merge branch 'V3/develop' of https://github.com/Cog-Creators/Red-DiscordBot into audiochanges

Drapersniper committed:
Remove PyCharm noise + Fixes a few Pycharm warnings

Signed-off-by: guyre <27962761+drapersniper@users.noreply.github.com>

Ianardo-DiCaprio committed:
Merge pull request #24 from Cog-Creators/V3/develop

t

Ianardo-DiCaprio committed:
Merge pull request #28 from Drapersniper/audiochanges

Audiochanges

Ianardo-DiCaprio committed:
Merge pull request #29 from Ianardo-DiCaprio/patch-1

Patch 1

Ianardo-DiCaprio committed:
Merge pull request #30 from Ianardo-DiCaprio/patch-2-1

Patch 2 1

Ianardo-DiCaprio committed:
Merge branch 'new' into patch-3-1

Ianardo-DiCaprio commented:
Are you fucking kidding me.... Fuck conflicted merges....

Drapersniper commented:
👀  You built this off my Music PR branch? may be worth mentioning that in the future just in case this was merged before the audio PR is
